src/quail/sensors/SensorTask.cpp

```diff
@@ -0,0 +1,40 @@
+#include "SensorTask.hpp"
+#include "main.hpp"
+
+StaticEventGroup_t SensorTask::evbuf;
+EventGroupHandle_t SensorTask::evgroup = xEventGroupCreateStatic(&evbuf);
+
+void SensorTask::activity(){
+    xEventGroupWaitBits(evgroup, ADC_STARTED, pdTRUE, pdFALSE, NEVER); // wait for ADC to begin (don't clear bits)
+    for(uint8_t i = 0; i < Sensor::numSensors(); i++){
+        sys.sensors[i]->configure();
+    }
+    sys.tasks.adctask.sensorsConfigured();
+    while(true){
+        uint32_t flags = xEventGroupWaitBits(evgroup, ADC_RESET|DATA_READY, pdTRUE, pdFALSE, NEVER); // await a reset or a data ready flag from ADC
+        if(flags & ADC_RESET)
+            xEventGroupWaitBits(evgroup, ADC_STARTED, pdTRUE, pdFALSE, NEVER);
```

Comment:
Does this properly handle reconfiguration? I think we need to break, and then I don't know if the sensor objects handle reconfiguring correctly.